eclipse-collections/src/main/java/org/eclipse/collections/impl/block/predicate/DropWhileIterablePredicate.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2016 Goldman Sachs.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v. 1.0 which accompany this distribution.
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ */
+
+package org.eclipse.collections.impl.block.predicate;
+
+import org.eclipse.collections.api.block.predicate.Predicate;
+
+public class DropWhileIterablePredicate<T> implements Predicate<T>
+{
+    private static final long serialVersionUID = 1L;
+
+    private final Predicate<? super T> predicate;
+    private boolean evaluate;
+
+    public DropWhileIterablePredicate(Predicate<? super T> predicate)
+    {
+        this.predicate = predicate;
+    }
+
+    public boolean accept(T each)
+    {
+        if (!this.evaluate && !this.predicate.accept(each))
+        {
+            this.evaluate = true;
+        }
+        return this.evaluate;
```

Comment:
It's used to determine whether we're done dropping elements and should *not* evaluate the predicate. We could keep the name evaluate if we negated every usage. How about doneDroppingElements?